docs_2.0/manual_EN/4.ngql-guide/12.vertex-statements/3.upsert-vertex.md

```diff
@@ -0,0 +1,50 @@
+# UPSERT VERTEX
+
+```nGQL
+UPSERT VERTEX <vid> SET <update_columns> [WHEN <condition>] [YIELD <columns>]
+```
+
+- `vid` is the ID of the vertex to be updated.
+- `update_columns` is the properties of the vertex to be updated. For example, `tag1.col1 = $^.tag2.col2 + 1` means to update `tag1.col1` to `tag2.col2+1`.
+
+    **NOTE:**  `$^` indicates the vertex to be updated.
+
+- `condition` is some constraints. Only when the conditions are met, `UPSERT` is executed successfully. `condition` supports expression operations.
+- `columns` is the columns to be returned, `YIELD` returns the latest updated values.
+
+`UPSERT` is a combination of `UPDATE` and `INSERT`. Use `UPSERT VERTEX` to update properties on a vertex or insert a new vertex. The `UPDATE VERTEX` statement only updates one tag of a vertex at a time.
+
+The performance of `UPSERT` is much lower than that of `INSERT`. Because `UPSERT` is a read-modify-write serialization operation at the partition level. So do not use `UPSERT` for large concurrent write scenarios.
+
+- If the vertex does not exist, a new vertex is created no matter the condition in the `WHEN` clause is met or not. The property columns not specified by the `SET` statement use the default values of the columns, if there are no default values, an error will be returned.
+- If the vertex exists and the `WHEN` condition is met, the vertex is updated.
+- If the vertex exists and the `WHEN` condition is not met, Nebula Graph does nothing.
+
+Consider the following example:
+
+```ngql
+nebula> INSERT VERTEX player(name, age) VALUES "player111":("Ben Simmons", 22); -- Insert a new vertex.
+nebula> UPSERT VERTEX "player111" SET player.name = "Dwight Howard", player.age = $^.player.age + 11 WHEN $^.player.name == "Ben Simmons" AND $^.player.age > 20 YIELD $^.player.name AS Name, $^.player.age AS Age; -- Do an upsert operation on the vertex.
+
++-----------------+-----+
+| Name            | Age |
++-----------------+-----+
+| "Dwight Howard" | 33  |
++-----------------+-----+
+```
+
+```ngql
+nebula> FETCH PROP ON * "player123"; -- An empty set is returned, indicating vertex 111 does not exist.
```

Comment:
indicating vertex 111 does not exist.==> "player123" ?

Reply:
nice catch